### ISSUE TITLE
Support building Unity Source Externally to Unity

### ISSUE THREAD

**Issue by andreiborodin** [microsoft/MSBuildForUnity#67]
There was an ask to have the following scenario:
- A Unity project that references an MSBuild project
- The MSBuild project compiles a source file that depends on Unity concept
  - This source file isn't compiled by Unity
- The output of this MSBuild is brought into Unity

Support for this could be as minimum as setting up a proper sample based on existing capabilities, or we can look into doing something more here.

Hacked up sample is #61, request originated in #44.

**Comment by loic-lopez** [comment]
Thanks for supporting this feature, I have another question: 

Do you think is it possible to have a directory next to the asset folder named for example `CustomUnityBuild` which references `UnityEngine` or other Unity namespaces?

example:
```
UnityProject\Assets\ CustomUnityBuild\.src
```
to ->
```
UnityProject\CustomUnityBuild
```

which output the dll in `UnityProject\Assets\Plugins\...`

**Comment by andreiborodin** [comment]
Yeah, you should be able to move the project in that example to anywhere under "UnityProject"
